src/entities/protocols/seaportV1_4.ts

```diff
@@ -0,0 +1,137 @@
+import abi from '../../../abis/SeaportV1_4.json'
+import { BigNumber } from 'ethers'
+import { Interface } from '@ethersproject/abi'
+import { BuyItem, Market, NFTTrade, TokenType } from '../NFTTrade'
+import { TradeConfig } from '../Command'
+import { RoutePlanner, CommandType } from '../../utils/routerCommands'
+import { ConsiderationItem, FulfillmentComponent, Order, SeaportData } from './seaport'
+
+export type AdvancedOrder = Order & {
+  numerator: BigNumber // uint120
+  denominator: BigNumber // uint120
+  extraData: string // bytes
+}
+
+export class SeaportV1_4Trade extends NFTTrade<SeaportData> {
```

Comment:
Copy and pasted from `./seaport.ts`, subbing out SeaportTrade with SeaportV1_4Trade and changing the CommandType to SeaportV1_4

Reply:
any way you can maybe make an abstract class in seaport.ts that both `SeaportTrade` and `SeaportV1_4Trade` both extend so we don't have so much code duplication?